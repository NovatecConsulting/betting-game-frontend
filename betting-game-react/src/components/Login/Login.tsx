import React from "react";

const Login: React.FC = () => (
    <div
        className="min-h-screen min-w-screen bg-white text-center text-6xl text-black grid items-center tracking-wider"
        data-testid="Login"
    >
        Login Component
    </div>
);

export default Login;
