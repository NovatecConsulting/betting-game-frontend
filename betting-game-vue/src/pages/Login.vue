<template>
  <div class="h-full w-screen grid items-center mt-0 md:mt-1/12">
    <form @submit.prevent="login">
      <div class="md:w-1/2 w-full md:px-16 py-8 px-4 shadow-xl rounded grid items-center md:mx-auto">
        <span class="my-12 text-4xl font-bold">Login</span>
        <input
          class="w-full h-12 bg-gray-200 px-10 mb-4 rounded-full hover:bg-gray-300 focus:outline-none focus:bg-white focus:border-blue-500 appearance-none outline-none border-2 border-gray-200"
          type="text"
          :placeholder="$t('loginpage.username.placeholder')"
          v-model="username"
          required
        />
        <input
          class="w-full h-12 bg-gray-200 px-10 mb-4 rounded-full hover:bg-gray-300 focus:outline-none focus:bg-white focus:border-blue-500 appearance-none outline-none border-2 border-gray-200"
          type="password"
          :placeholder="$t('loginpage.password.placeholder')"
          v-model="password"
          required
        />
        <div class="buttons flex">
          <button
            type="submit"
            class="buttons__login px-8 h-12 rounded-full bg-blue-500 hover:bg-blue-600 text-white font-bold outline-none select-none mr-4 focus:outline-none"
          >
            {{ $t('loginpage.login.button') }}
          </button>
          <button
            type="button"
            class="buttons__forgot px-8 h-12 rounded-full bg-background font-bold border-2 border-secondary focus:outline-none"
          >
            {{ $t('loginpage.forgotpassword.button') }}
          </button>
        </div>
      </div>
    </form>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import { vxm } from '../store/store.vuex'

@Component
export default class Login extends Vue {
  username = ''
  password = ''
  user = vxm.user

  async login() {
    await this.user.login({
      username: this.username,
      password: this.password
    })

    if (this.user.isLoggedIn) {
      this.$router.push('/')
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped></style>
