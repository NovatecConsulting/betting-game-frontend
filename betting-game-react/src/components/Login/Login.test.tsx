import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Login from './Login';

describe('<Login />', () => {
  test('it should mount', () => {
    render(<Login />);

    const login = screen.getByTestId('Login');

    expect(login).toBeInTheDocument();
  });
});
