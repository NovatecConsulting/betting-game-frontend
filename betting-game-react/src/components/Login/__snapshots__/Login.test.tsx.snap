// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should render and match snapshot with correct content text. 1`] = `
<div
  className="min-h-screen min-w-screen bg-white text-center text-6xl text-black grid items-center tracking-wider"
  data-testid="Login"
>
  Login Component
</div>
`;
